src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/DataElementsCard.tsx

```diff
@@ -25,7 +29,91 @@ const DataElementsCard = (props: {
   const negationRationale =
     selectedDataElement?.hasOwnProperty("negationRationale");
   // https://ecqi.healthit.gov/mcw/2020/qdm-attribute/negationrationale.html  (list of all categories that use negation rationale)
-
+  const formik: any = useFormikContext();
+  // from here we know the type, we need to go through the dataElements to matchTypes
+  // attributes section
+  const [displayAttributes, setDisplayAttributes] = useState([]);
+  // codes section
+  const [codesChips, setCodesChips] = useState([]);
+  useEffect(() => {
+    if (formik.values?.json && selectedDataElement) {
+      const codesChips = [];
+      let patient = null;
+      patient = JSON.parse(formik.values.json);
```

Comment:
as discussed, we'd want to only display the attributes associated with the currently selected data element, which means there should be no need to parse the testcase json